From 9fd361108ec3f19b6078b02e31e84fd7ae41c557 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 8 Oct 2019 20:45:34 +0200
Subject: [PATCH 1/1] arm: remove the H2200 board

U-Boot cannot be built for h2200_defconfig with CONFIG_DM=y.

The maintainer Lukasz Dalek suggested to remove the board.
https://lists.denx.de/pipermail/u-boot/2019-August/380685.html

Cc: Lukasz Dalek <luk0104@gmail.com>
Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 arch/arm/Kconfig           |   4 --
 board/h2200/Kconfig        |   9 ---
 board/h2200/MAINTAINERS    |   6 --
 board/h2200/Makefile       |  12 ----
 board/h2200/h2200-header.S |  14 ----
 board/h2200/h2200.c        |  73 ---------------------
 configs/h2200_defconfig    |  38 -----------
 include/configs/h2200.h    | 128 -------------------------------------
 8 files changed, 284 deletions(-)
 delete mode 100644 board/h2200/Kconfig
 delete mode 100644 board/h2200/MAINTAINERS
 delete mode 100644 board/h2200/Makefile
 delete mode 100644 board/h2200/h2200-header.S
 delete mode 100644 board/h2200/h2200.c
 delete mode 100644 configs/h2200_defconfig
 delete mode 100644 include/configs/h2200.h

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 3b0e315061..416d0fdb75 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -1479,9 +1479,6 @@ config TARGET_LS1046AFRWY
 	  The LS1046A Freeway Board (FRWY) is a high-performance
 	  development platform that supports the QorIQ LS1046A
 	  Layerscape Architecture processor.
-config TARGET_H2200
-	bool "Support h2200"
-	select CPU_PXA
 
 config TARGET_COLIBRI_PXA270
 	bool "Support colibri_pxa270"
@@ -1789,7 +1786,6 @@ source "board/freescale/mx35pdk/Kconfig"
 source "board/freescale/s32v234evb/Kconfig"
 source "board/grinn/chiliboard/Kconfig"
 source "board/gumstix/pepper/Kconfig"
-source "board/h2200/Kconfig"
 source "board/hisilicon/hikey/Kconfig"
 source "board/hisilicon/hikey960/Kconfig"
 source "board/hisilicon/poplar/Kconfig"
diff --git a/board/h2200/Kconfig b/board/h2200/Kconfig
deleted file mode 100644
index c0e0c1e763..0000000000
--- a/board/h2200/Kconfig
+++ /dev/null
@@ -1,9 +0,0 @@
-if TARGET_H2200
-
-config SYS_BOARD
-	default "h2200"
-
-config SYS_CONFIG_NAME
-	default "h2200"
-
-endif
diff --git a/board/h2200/MAINTAINERS b/board/h2200/MAINTAINERS
deleted file mode 100644
index b66ff515cf..0000000000
--- a/board/h2200/MAINTAINERS
+++ /dev/null
@@ -1,6 +0,0 @@
-H2200 BOARD
-M:	Lukasz Dalek <luk0104@gmail.com>
-S:	Maintained
-F:	board/h2200/
-F:	include/configs/h2200.h
-F:	configs/h2200_defconfig
diff --git a/board/h2200/Makefile b/board/h2200/Makefile
deleted file mode 100644
index 690b766178..0000000000
--- a/board/h2200/Makefile
+++ /dev/null
@@ -1,12 +0,0 @@
-# SPDX-License-Identifier: GPL-2.0+
-#
-# h2200 Support
-#
-# Copyright (C) 2012 Lukasz Dalek <luk0104@gmail.com>
-
-obj-y	:= h2200.o
-
-extra-y := h2200-header.bin
-
-$(obj)/h2200-header.bin: $(obj)/h2200-header.o
-	$(OBJCOPY) -O binary $< $@
diff --git a/board/h2200/h2200-header.S b/board/h2200/h2200-header.S
deleted file mode 100644
index be8b7fb923..0000000000
--- a/board/h2200/h2200-header.S
+++ /dev/null
@@ -1,14 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0+ */
-/*
- * iPAQ h2200 header
- *
- * Copyright (C) 2012 Lukasz Dalek <luk0104@gmail.com>
- */
-
-	.word 0xea0003fe /* b 0x1000 */
-
-	.org 0x40
-	.ascii "ECEC"
-
-	.org 0x1000 - 1
-	.byte 0x0
diff --git a/board/h2200/h2200.c b/board/h2200/h2200.c
deleted file mode 100644
index e1b7b2ce8a..0000000000
--- a/board/h2200/h2200.c
+++ /dev/null
@@ -1,73 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0+
-/*
- * iPAQ h2200 board configuration
- *
- * Copyright (C) 2012 Lukasz Dalek <luk0104@gmail.com>
- */
-
-#include <common.h>
-#include <asm/arch/pxa.h>
-#include <asm/arch/pxa-regs.h>
-#include <asm/io.h>
-#include <usb.h>
-#include <asm/mach-types.h>
-
-DECLARE_GLOBAL_DATA_PTR;
-
-int board_eth_init(bd_t *bis)
-{
-	usb_eth_initialize(bis);
-	return 0;
-}
-
-void reset_cpu(ulong ignore)
-{
-	/* Enable VLIO interface on Hamcop */
-	writeb(0x1, 0x4000);
-
-	/* Reset board (cold reset) */
-	writeb(0xff, 0x4002);
-}
-
-int board_init(void)
-{
-	/* We have RAM, disable cache */
-	dcache_disable();
-	icache_disable();
-
-	gd->bd->bi_arch_number = MACH_TYPE_H2200;
-
-	/* adress of boot parameters */
-	gd->bd->bi_boot_params = 0xa0000100;
-
-	/* Let host see that device is disconnected */
-	udc_disconnect();
-	mdelay(500);
-
-	return 0;
-}
-
-int dram_init(void)
-{
-	/*
-	 * Everything except MSC0 was already set up by
-	 * 1st stage bootloader.
-	 *
-	 * This setting enables access to companion chip.
-	 */
-	clrsetbits_le32(MSC0, 0xffffffff, CONFIG_SYS_MSC0_VAL);
-	gd->ram_size = CONFIG_SYS_SDRAM_SIZE;
-	return 0;
-}
-
-#ifdef CONFIG_USB_GADGET_PXA2XX
-int board_usb_init(int index, enum usb_init_type init)
-{
-	return 0;
-}
-
-int board_usb_cleanup(int index, enum usb_init_type init)
-{
-	return 0;
-}
-#endif
diff --git a/configs/h2200_defconfig b/configs/h2200_defconfig
deleted file mode 100644
index 72bb4d3f70..0000000000
--- a/configs/h2200_defconfig
+++ /dev/null
@@ -1,38 +0,0 @@
-CONFIG_ARM=y
-CONFIG_TARGET_H2200=y
-CONFIG_SYS_TEXT_BASE=0xa0041000
-CONFIG_NR_DRAM_BANKS=1
-CONFIG_FIT=y
-# CONFIG_FIT_ENABLE_SHA256_SUPPORT is not set
-CONFIG_USE_BOOTARGS=y
-CONFIG_BOOTARGS="root=/dev/ram0 ro console=ttyS0,115200n8"
-CONFIG_SYS_CONSOLE_IS_IN_ENV=y
-# CONFIG_DISPLAY_CPUINFO is not set
-# CONFIG_DISPLAY_BOARDINFO is not set
-CONFIG_HUSH_PARSER=y
-# CONFIG_CMDLINE_EDITING is not set
-# CONFIG_AUTO_COMPLETE is not set
-# CONFIG_SYS_LONGHELP is not set
-CONFIG_SYS_PROMPT="> "
-# CONFIG_CMD_BDI is not set
-# CONFIG_CMD_CONSOLE is not set
-# CONFIG_CMD_BOOTD is not set
-# CONFIG_CMD_XIMG is not set
-# CONFIG_CMD_EDITENV is not set
-# CONFIG_CMD_SAVEENV is not set
-# CONFIG_CMD_ENV_EXISTS is not set
-# CONFIG_CMD_MEMORY is not set
-# CONFIG_CMD_FLASH is not set
-# CONFIG_CMD_LOADS is not set
-# CONFIG_CMD_ECHO is not set
-# CONFIG_CMD_ITEST is not set
-# CONFIG_CMD_SETEXPR is not set
-# CONFIG_CMD_NFS is not set
-CONFIG_CMD_PING=y
-# CONFIG_CMD_MISC is not set
-# CONFIG_MMC is not set
-CONFIG_CONS_INDEX=3
-CONFIG_PXA_SERIAL=y
-CONFIG_USB=y
-CONFIG_USB_GADGET=y
-CONFIG_USB_ETHER=y
diff --git a/include/configs/h2200.h b/include/configs/h2200.h
deleted file mode 100644
index 1abf283e26..0000000000
--- a/include/configs/h2200.h
+++ /dev/null
@@ -1,128 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0+ */
-/*
- * iPAQ h2200 board configuration
- *
- * Copyright (C) 2012 Lukasz Dalek <luk0104@gmail.com>
- */
-
-#ifndef __CONFIG_H
-#define __CONFIG_H
-
-#define CONFIG_MACH_TYPE		MACH_TYPE_H2200
-
-#define CONFIG_CPU_PXA25X		1
-
-#define PHYS_SDRAM_1			0xa0000000 /* SDRAM Bank #1 */
-#define PHYS_SDRAM_1_SIZE		0x04000000 /* 64 MB */
-
-#define CONFIG_SYS_SDRAM_BASE		PHYS_SDRAM_1
-#define CONFIG_SYS_SDRAM_SIZE		PHYS_SDRAM_1_SIZE
-
-#define CONFIG_SYS_INIT_SP_ADDR		0xfffff800
-
-#define CONFIG_ENV_SIZE			0x00040000
-#define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + 128*1024)
-
-#define CONFIG_SYS_LOAD_ADDR		0xa3000000 /* default load address */
-
-/*
- * iPAQ 1st stage bootloader loads 2nd stage bootloader
- * at address 0xa0040000 but bootloader requires header
- * which is 0x1000 long.
- *
- * --- Header begin ---
- *	.word 0xea0003fe ; b 0x1000
- *
- *	.org 0x40
- *	.ascii "ECEC"
- *
- *	.org 0x1000
- * --- Header end ---
- */
-
-/*
- * Static chips
- */
-
-#define CONFIG_SYS_MSC0_VAL		0x246c7ffc
-#define CONFIG_SYS_MSC1_VAL		0x7ff07ff0
-#define CONFIG_SYS_MSC2_VAL		0x7ff07ff0
-
-/*
- * PCMCIA and CF Interfaces
- */
-
-#define CONFIG_SYS_MECR_VAL		0x00000000
-#define CONFIG_SYS_MCMEM0_VAL		0x00000000
-#define CONFIG_SYS_MCMEM1_VAL		0x00000000
-#define CONFIG_SYS_MCATT0_VAL		0x00000000
-#define CONFIG_SYS_MCATT1_VAL		0x00000000
-#define CONFIG_SYS_MCIO0_VAL		0x00000000
-#define CONFIG_SYS_MCIO1_VAL		0x00000000
-
-#define CONFIG_SYS_FLYCNFG_VAL		0x00000000
-#define CONFIG_SYS_SXCNFG_VAL		0x00040004
-
-#define CONFIG_SYS_MDREFR_VAL		0x0099E018
-#define CONFIG_SYS_MDCNFG_VAL		0x01C801CB
-#define CONFIG_SYS_MDMRS_VAL		0x00220022
-
-#define CONFIG_SYS_PSSR_VAL		0x00000000
-#define CONFIG_SYS_CKEN			0x00004840
-#define CONFIG_SYS_CCCR			0x00000161
-
-/*
- * GPIOs
- */
-
-#define CONFIG_SYS_GPSR0_VAL		0x01000000
-#define CONFIG_SYS_GPSR1_VAL		0x00000000
-#define CONFIG_SYS_GPSR2_VAL		0x00010000
-
-#define CONFIG_SYS_GPCR0_VAL		0x00000000
-#define CONFIG_SYS_GPCR1_VAL		0x00000000
-#define CONFIG_SYS_GPCR2_VAL		0x00000000
-
-#define CONFIG_SYS_GPDR0_VAL		0xF7E38C00
-#define CONFIG_SYS_GPDR1_VAL		0xBCFFBF83
-#define CONFIG_SYS_GPDR2_VAL		0x000157FF
-
-#define CONFIG_SYS_GAFR0_L_VAL		0x80401000
-#define CONFIG_SYS_GAFR0_U_VAL		0x00000112
-#define CONFIG_SYS_GAFR1_L_VAL		0x600A9550
-#define CONFIG_SYS_GAFR1_U_VAL		0x0005AAAA
-#define CONFIG_SYS_GAFR2_L_VAL		0x20000000
-#define CONFIG_SYS_GAFR2_U_VAL		0x00000000
-
-/*
- * Serial port
- */
-#define CONFIG_FFUART
-
-#define CONFIG_SYS_BAUDRATE_TABLE	{ 9600, 38400, 115200 }
-
-#define CONFIG_SETUP_MEMORY_TAGS
-#define CONFIG_CMDLINE_TAG
-#define CONFIG_INITRD_TAG
-
-/* Monitor Command Prompt */
-
-#define CONFIG_USB_DEV_PULLUP_GPIO	33
-/* USB VBUS GPIO 3 */
-
-#define CONFIG_BOOTCOMMAND		\
-	"setenv downloaded 0 ; while test $downloaded -eq 0 ; do " \
-	"if bootp ; then setenv downloaded 1 ; fi ; done ; " \
-	"source :script ; " \
-	"bootm ; "
-
-#define CONFIG_USB_GADGET_PXA2XX
-#define CONFIG_USB_ETH_SUBSET
-
-#define CONFIG_USBNET_DEV_ADDR		"de:ad:be:ef:00:01"
-#define CONFIG_EXTRA_ENV_SETTINGS \
-	"stdin=serial\0" \
-	"stdout=serial\0" \
-	"stderr=serial\0"
-
-#endif /* __CONFIG_H */
-- 
2.23.0

